packages/react/src/hooks/useReverification.ts

```diff
@@ -0,0 +1,146 @@
+import { reverificationMismatch } from '@clerk/shared/authorization-errors';
+import { ClerkRuntimeError, isClerkAPIResponseError, isClerkRuntimeError } from '@clerk/shared/error';
+import { useClerk } from '@clerk/shared/react';
+import type { __experimental_ReverificationConfig, Clerk } from '@clerk/types';
+import { useMemo } from 'react';
+
+interface PromiseWithResolvers<T> {
+  promise: Promise<T>;
+  resolve: (value: T | PromiseLike<T>) => void;
+  reject: (reason?: any) => void;
+}
+
+async function resolveResult<T>(result: Promise<T>): Promise<T | ReturnType<typeof reverificationMismatch>> {
+  return result
+    .then(r => {
+      if (r instanceof Response) {
+        return r.json();
+      }
+      return r;
+    })
+    .catch(e => {
+      // Treat fapi assurance as an assurance hint
+      if (isClerkAPIResponseError(e) && e.errors.find(({ code }) => code == 'session_step_up_verification_required')) {
+        return reverificationMismatch({} as __experimental_ReverificationConfig);
```

Comment:
❓ Why do we need the empty object here?

Reply:
i've updated the signature to allow for undefined